weather-oracle/sources/weather.move

```diff
@@ -0,0 +1,377 @@
+                latitude,  // Set the latitude of the city weather oracle object
+                positive_latitude,  // Set whether the latitude is positive (north) or negative (south)
+                longitude,  // Set the longitude of the city weather oracle object
+                positive_longitude,  // Set whether the longitude is positive (east) or negative (west)
+                weather_id: 0, // Initialize the weather condition code to be zero 
+                temp: 0, // Initialize the temperature to be zero 
+                pressure: 0, // Initialize the pressure to be zero 
+                humidity: 0, // Initialize the humidity to be zero 
+                visibility: 0, // Initialize the visibility to be zero 
+                wind_speed: 0, // Initialize the wind speed to be zero 
+                wind_deg: 0, // Initialize the wind direction to be zero 
+                wind_gust: option::none(), // Initialize the wind gust to be none 
+                clouds: 0, // Initialize the cloudiness to be zero 
+                dt: 0 // Initialize the timestamp to be zero 
+            }
+        );
+    }
+
+    // Public function for removing an existing city from the oracle
+    public fun remove_city(_: &AdminCap, oracle: &mut WeatherOracle, geoname_id: u32) {
+        let CityWeatherOracle { id, geoname_id: _, name: _, country: _, latitude: _, positive_latitude: _, longitude: _, positive_longitude: _, weather_id: _, temp: _, pressure: _, humidity: _, visibility: _, wind_speed: _, wind_deg: _, wind_gust: _, clouds: _, dt: _ } = dof::remove(&mut oracle.id, geoname_id);
+        object::delete(id);
+    }
+
+    // Public function for updating the weather conditions of a city
+    public fun update(
+        _: &AdminCap,
+        oracle: &mut WeatherOracle,
+        geoname_id: u32,
+        weather_id: u16,
+        temp: u32,
+        pressure: u32,
+        humidity: u8,
+        visibility: u16,
+        wind_speed: u16,
+        wind_deg: u16,
+        wind_gust: Option<u16>,
+        clouds: u8,
+        dt: u32
+    ) {
+        let city_weather_oracle_mut = dof::borrow_mut<u32, CityWeatherOracle>(&mut oracle.id, geoname_id); // Borrow a mutable reference to the city weather oracle object with the geoname ID as the key
+        city_weather_oracle_mut.weather_id = weather_id;
+        city_weather_oracle_mut.temp = temp;
+        city_weather_oracle_mut.pressure = pressure;
+        city_weather_oracle_mut.humidity = humidity;
+        city_weather_oracle_mut.visibility = visibility;
+        city_weather_oracle_mut.wind_speed = wind_speed;
+        city_weather_oracle_mut.wind_deg = wind_deg;
+        city_weather_oracle_mut.wind_gust = wind_gust;
+        city_weather_oracle_mut.clouds = clouds;
+        city_weather_oracle_mut.dt = dt;
+    }
+
+    // Define a public function for minting an NFT that represents the current weather conditions of a city.
+    public fun mint(
```

Comment:
Should we consider allowing `burn` for these? For storage rebates?

Overall, I am curious why do we need these to be objects (having `key`)? What use cases do these bring?

Reply:
Adding a `burn` function for storage rebates is a really good idea. The contracts have already been deployed and used on the mainnet, but I will include a `burn` function in the next package upgrade.
We added it as an extra feature, but it is not used yet. One potential use case is that someone can get the data of a city as an NFT when they mint a new suifren and based on that NFT (think about this us an accessory) will render the suifre differently